src/main/java/de/flapdoodle/embed/process/config/store/FileSet.java

```diff
@@ -83,50 +83,83 @@ public boolean equals(Object other) {
 				return true;
 			}
 
-			if (other == null || getClass() != other.getClass()) {
+			if (other == null) { // || getClass() != other.getClass()) {
+				return false;
+			}
+
+			/* Preferable to checking class instance equality guard. See #43. */
+			if (!(other instanceof FileSet.Entry)) {
 				return false;
 			}
 
 			Entry entry = (Entry) other;
 
+			/* Because Pattern doesn't implement equals, use its underlying expression (String) for equality. See #43. */
 			return _type == entry._type
 				&& !(_destination != null ? !_destination.equals(entry._destination) : entry._destination != null)
-				&& !(_matchingPattern != null ? !_matchingPattern.equals(entry._matchingPattern) : entry._matchingPattern != null);
+				&& !(_matchingPattern != null ? !_matchingPattern.pattern().equals(null == entry._matchingPattern ? null : entry._matchingPattern.pattern()) : entry._matchingPattern != null);
```

Comment:
i know the former line was also like that but could you please flip operators here?